sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala

```diff
@@ -1484,4 +1484,27 @@ class StatisticsSuite extends StatisticsCollectionTestBase with TestHiveSingleto
       }
     }
   }
+
+  test("fallBackToHdfs should not support Hive partitioned table") {
+    Seq(true, false).foreach { fallBackToHdfs =>
+      withSQLConf(SQLConf.ENABLE_FALL_BACK_TO_HDFS_FOR_STATS.key -> s"$fallBackToHdfs",
+        HiveUtils.CONVERT_METASTORE_PARQUET.key -> "false") {
+        withTempDir { tempDir =>
+          withTable("spark_28876") {
+            sql(s"""
+               |CREATE EXTERNAL TABLE spark_28876(id bigint)
+               |PARTITIONED BY (ds STRING)
+               |STORED AS PARQUET
+               |LOCATION '${tempDir.toURI}'
+             """.stripMargin)
+            sql("INSERT INTO TABLE spark_28876 PARTITION (ds='p1') SELECT 1")
```

Comment:
can we use `ADD PARTITION` to test external partitions？

Reply:
Done